CONTRIBUTING.md

```diff
@@ -5,16 +5,161 @@
 
 Interested in contributing to LlamaIndex? Here's how to get started! 
 
-## Contributions that we're looking for:
-- Bug fixes
-- New features
+## Contribution Guideline
+The best part of LlamaIndex is our community of users and contributors.
 
-All future tasks are tracked in [Github Issues Page](https://github.com/jerryjliu/gpt_index/issues).
+
+### What should I work on?
+- 🆕 Extend core modules
+- 🐛 Fix bugs
+- 🎉 Add usage examples
+- 🧪 Add experimental features 
+- 📄 Improve code quality & documentation
+
+Also, join our Discord for ideas and discussions: https://discord.gg/dGcwcsnxhU.
+
+
+### 🆕 Extend Core Modules
+The most impactful way to contribute to LlamaIndex is extending our core modules:
+![LlamaIndex modules](docs/_static/contribution/contrib.png)
+
+
+We can roughly group the core modules into 3 categories:
+1) data ingestion & parsing, 
+2) indexing & storage, and 
+3) querying & output.
+
+We welcome contributions in _all_ modules shown above.
+So far, we have focused on implementing a core set of functionalities for each.
+As a contributor, you can help unlock the full potential.
+
+We are making rapid improvement to the project, and as a result, 
```

Comment:
*rapid improvements

Reply:
also would probably make this entire section one paragraph, and preface with **NOTE** (not sure if you meant entire paragraph to be quote text)